.github/workflows/integration_tests.yml

```diff
@@ -37,9 +37,17 @@ jobs:
     runs-on: ${{ needs.start-runner.outputs.label }}
     timeout-minutes: 600
     steps:
+      - name: Determine branch ref
+        id: branch_ref
+        run: |
+          if [ "${{ github.event_name }}" = "pull_request" ]; then
+            echo "branch_ref=refs/heads/${{ github.head_ref }}" >> $GITHUB_OUTPUT
+          else
+            echo "branch_ref=${{ github.ref }}" >> $GITHUB_OUTPUT
+          fi
```

Comment:
This step creates a new output variable which determines what the branch_ref (fancy name for branch name) will be. The goal is to create the variable that will evaluate branch_ref to `refs/heads/<name of the branch>`.